includes/Story_Post_Type.php

```diff
@@ -340,4 +348,59 @@ public function update_archive_setting() {
 			flush_rewrite_rules( false );
 		}
 	}
+
+	/**
+	 * Determines whether the post type should have an archive or not.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @return bool|string Whether the post type should have an archive, or archive slug.
+	 */
+	private function get_has_archive() {
+		$archive_page_option    = get_option( Settings::SETTING_NAME_ARCHIVE );
+		$custom_archive_page_id = get_option( Settings::SETTING_NAME_ARCHIVE_PAGE_ID );
+		$has_archive            = true;
+
+		if ( 'disabled' === $archive_page_option ) {
+			$has_archive = false;
+		} elseif ( 'custom' === $archive_page_option && $custom_archive_page_id ) {
+			$uri = get_page_uri( $custom_archive_page_id );
+			if ( $uri ) {
+				$has_archive = urldecode( $uri );
+			}
+		}
+
+		return $has_archive;
+	}
+
+	/**
+	 * Modifies the current query to set up the custom archive page.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_Query $query Current query instance, passed by reference.
+	 * @return void
+	 */
+	public function pre_get_posts( WP_Query $query ) {
+		if ( ! is_string( $this->get_has_archive() ) ) {
+			return;
+		}
+
+		if ( $query->is_admin || ! $query->is_main_query() ) {
+			return;
+		}
+
+		if ( ! $query->is_post_type_archive( self::POST_TYPE_SLUG ) ) {
+			return;
+		}
+
+		$custom_archive_page_id = get_option( Settings::SETTING_NAME_ARCHIVE_PAGE_ID );
```

Comment:
What happens here if you delete the page?

Reply:
What if the post type is trashed or draft?

Reply:
[get_page_uri](https://developer.wordpress.org/reference/functions/get_page_uri/) doesn't seem to have a check for post status?

Reply:
Good point. Added a check for post status now.